awx/main/utils/ansible.py

```diff
@@ -92,8 +95,13 @@ def could_be_inventory(project_path, dir_path, filename):
             errors='ignore'
         ) as inv_file:
             for line in islice(inv_file, 10):
+                if 'plugin:' in line or 'all' in line:
+                    # Probably an inventory file
+                    return inventory_rel_path
                 if not valid_inventory_re.match(line):
                     return None
+            else:
+                return None
```

Comment:
This would fail to include files such as

https://github.com/ansible/awx/blob/devel/awx/main/tests/data/inventory/plugins/ec2/files/aws_ec2.yml

Because the `plugin:` keyword is not in the first 10 lines of the file.

the regex IIRC is fairly specifically targeted toward ini-type files (including those that don't actually have the .ini extension explicitly). Moving forward, the only thing that makes sense to me is to break this logic up for different extension types (or lack thereof). I would still prefer to avoid doing a `json.loads` on all YAML files in source due to performance concerns. Maybe larger YAML files should just get included by default. Maybe we specifically exclude only YAML files that are under 10 lines and don't have the "plugin:" syntax.

Reply:
I'd not noticed that there was such a large inventory plugin file there! It seems weird to me that the plugin key isn't at the beginning, but sure, there's no reason why it can't be at the end. My idea was to make just a tiny nudge to help improve the situation without making a large adjustment, and not expecting to catch everything. But I am ok with making this a deeper change. I'll update the PR.

Regardless, this isn't the best solution, as the UI still limits the total results to 50 (and I have a project with >1000 inventory files!) I think the best way to handle it is the same way we pick projects -- a modal box that is filterable, and additionally, able to drill down into directories. But I don't think I can spend the time to implement that.

Reply:
> It seems weird to me that the plugin key isn't at the beginning, but sure, there's no reason why it can't be at the end.

The standard python YAML library sorts keys alphabetically, so that's why in this case.

@jlmitch5 might have some opinions on the UI comments. I think the API itself has some limit. I don't think playbook / inventory files I don't think have ever had filtering applied to them, like with the related items you mention.